src/ol/interaction/Modify.js

```diff
@@ -1016,10 +1016,10 @@ class Modify extends PointerInteraction {
     if (this.lastPointerEvent_ && this.lastPointerEvent_.type != MapBrowserEventType.POINTERDRAG) {
       const evt = this.lastPointerEvent_;
       this.willModifyFeatures_(evt);
-      this.removeVertex_();
+      const removed = this.removeVertex_();
       this.dispatchEvent(new ModifyEvent(ModifyEventType.MODIFYEND, this.features_, evt));
```

Comment:
You could displatch this event only when the vertex was removed:
```js
if (removed) {
  this.dispatchEvent(new ModifyEvent(ModifyEventType.MODIFYEND, this.features_, evt));
}
```

Reply:
Another thing I just thought of - it might be beneficial to have a public API to check whether a vertex can be removed. This could, for example, be used to hide a "remove vertex"-button in the UI if removal currently isn't possible.

Reply:
It could look something like [this](https://github.com/ludvigeriksson/openlayers/commit/c7154b1810cc99c194a2a9495566685f6dd9a9c1). It might need more thorough testing before merging, but it shows my thinking.

Reply:
I think for the sake of library code simplicity, we'll leave things as in this pull request and let the user deal with modifystart/end events that do not result in any modification. Ok for you @ludvigeriksson?

Reply:
Sure thing